spec/controllers/v2/concerns/collection_spec.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+# The `parents` parameter is required when testing nested controllers, and it should contain
+# an ordered list of reflection symbols similarly to how they are defined in the routes. It is
+# also required to set the `extra_params` variable in a let block and pass all the parent IDs
+# there as a hash. For example:
+# ```
+# it_behaves_like 'collection', :security_guide, :profile do
+#   let(:extra_params) { { security_guide_id: 123, profile_id: 456 } }
+# end
+# ```
+#
+# In a non-nested case this example can be simply included as:
+# ```
+# it_behaves_like 'collection'
+# ```
+#
+RSpec.shared_examples 'collection' do |*parents|
+  let(:extra_params) { {} }
+
+  it 'returns base fields for each result' do
+    collection = items.map do |item|
+      hash_including(
+        'id' => item.id,
+        'type' => described_class.controller_name.chop,
+        **attributes.each_with_object({}) do |(key, value), obj|
+          obj[key.to_s] = item.send(value)
+        end
+      )
+    end
+
+    get :index, params: extra_params.merge(parents: parents)
+
+    expect(response).to have_http_status :ok
+    expect(response_body_data).to match_array(collection)
+
+    response_body_data.each do |item|
+      expect(item.keys.count).to eq(attributes.keys.count + 2)
+    end
+  end
+
+  context 'under incorrect parent security guide', if: parents.present? do
```

Comment:
I didn't even know this syntax existed 😁 not sure if this would look better, I'll think about it
```suggestion
  if parents.present?
    context 'under incorrect parent security guide' do
```